ui/src/app/components/apps-panel.tsx

```diff
@@ -0,0 +1,413 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import {
+  faGear,
+  faPause,
+  faPlay,
+  faPlusCircle,
+  faTrashCan,
+} from '@fortawesome/free-solid-svg-icons';
+import { faSyncAlt } from '@fortawesome/free-solid-svg-icons/faSyncAlt';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { BillingStatus, RuntimeStatus, Workspace } from 'generated/fetch';
+
+import { Clickable } from 'app/components/buttons';
+import { FlexColumn, FlexRow } from 'app/components/flex';
+import { DisabledPanel } from 'app/components/runtime-configuration-panel/disabled-panel';
+import { RuntimeStatusIcon } from 'app/components/runtime-status-icon';
+import { NewNotebookModal } from 'app/pages/analysis/new-notebook-modal';
+import { dropNotebookFileSuffix } from 'app/pages/analysis/util';
+import { notebooksApi } from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import { cond, reactStyles, switchCase } from 'app/utils';
+import { machineRunningCost, machineStorageCost } from 'app/utils/machines';
+import { formatUsd } from 'app/utils/numbers';
+import {
+  isActionable,
+  isVisible,
+  RuntimeStatusRequest,
+  toAnalysisConfig,
+  useRuntimeStatus,
+} from 'app/utils/runtime-utils';
+import {
+  diskStore,
+  runtimeStore,
+  serverConfigStore,
+  useStore,
+} from 'app/utils/stores';
+import jupyterLogo from 'assets/images/jupyter.png';
+import rStudioLogo from 'assets/images/RStudio.png';
+
+const styles = reactStyles({
+  expandedAppContainer: {
+    background: colors.white,
+    marginLeft: '1em',
+    marginBottom: '1em',
+    padding: '1em',
+    width: 'fit-content',
+  },
+  button: {
+    opacity: 1,
+    boxSizing: 'border-box',
+    height: 69,
+    width: 108,
+    border: '1px solid #979797',
+    borderRadius: 9,
+    alignItems: 'center',
+  },
+  disabledButton: {
+    opacity: 0.46,
+    cursor: 'not-allowed',
+    boxSizing: 'border-box',
+    height: 69,
+    width: 108,
+    border: '1px solid #979797',
+    borderRadius: 9,
+    alignItems: 'center',
+  },
+  enabledTrashButton: {
+    alignSelf: 'center',
+    marginLeft: 'auto',
+    marginRight: '1em',
+  },
+  disabledTrashButton: {
+    alignSelf: 'center',
+    marginLeft: 'auto',
+    marginRight: '1em',
+    color: colors.disabled,
+    cursor: 'not-allowed',
+  },
+  buttonText: {
+    color: '#4D72AA', // TODO fix
+    fontFamily: 'Montserrat',
+    fontWeight: 'bold',
+    letterSpacing: 0,
+    lineHeight: '15px',
+  },
+  buttonIcon: {
+    height: 23,
+    padding: '0.7em',
+  },
+  header: {
+    color: colors.primary,
+    height: 19,
+    fontFamily: 'Montserrat',
+    fontSize: 16,
+    fontWeight: 600,
+    letterSpacing: 0,
+    lineHeight: '19px',
+    margin: '1em',
+  },
+  availableApp: {
+    background: colors.white,
+    marginLeft: '1em',
+    marginBottom: '1em',
+    justifyContent: 'center',
+  },
+});
+
+// Eventually we will need to align this with the API's AppType
```

Comment:
More extensive.  For one thing, we are likely to have stages in the rollout process where the backend supports more apps than the UI does - and different parts of the UI may differ as well.  For now, this is just hardcoding what *this piece of the UI* can do.

Another important point is that Jupyter is **not an app** from the backend's perspective, because **runtimes aren't apps** - but eventually they will be.

It's going to messy, and I don't think today's guesses will match the final shape of things.  So let's defer this until later.